### PR TITLE
Workaround for visually missing health on join

### LINKED ISSUE #4793: Visual glitch with custom hearts container
### What behaviour is expected:
If you have custom heart containers (example 40 not the default 20) when you join you can see all containers filled.

### What behaviour is observed:
When join (you have full life) the client only show 20 hearts filled and the another 20 not filled, if you take damaage then the client reload the rest of containers full.

### Steps/models to reproduce:
1. Join to the server
2. Run this command `/attribute @s minecraft:generic.max_health base set 40`
3. Get all the containers filled
4. Disconnect from server
5. Reconnect to Server
6. Take damage

### Plugin list:
AsyncWorldEdit, ConterCraft, CoreProtect, LibsDisguises, LuckPerms, PlugMan, ProtocolLib, WorldEdit

### Paper version:
git-Paper-288 (MC: 1.16.4) (Implementing API version 1.16.4-R0.1-SNAPSHOT)

### Anything else:
Gif of issue: https://imgur.com/rp8sWqA

### PR DESCRIPTION
**Pull request by HexedHero**
Force updates the player's health after they join

Fixes #4793 and [SPIGOT-6250](https://hub.spigotmc.org/jira/browse/SPIGOT-6250)
Before: https://i.imgur.com/z16npIM.mp4
After: https://i.imgur.com/b5RBlnN.mp4

### CODE REVIEW

File: Spigot-Server-Patches/0662-Workaround-for-visually-missing-health-on-join.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: HexedHero <6012891+HexedHero@users.noreply.github.com>
```
Comment (Proximyst) [inline Spigot-Server-Patches/0662-Workaround-for-visually-missing-health-on-join.patch]:
```suggestion
From: Doc <nachito94@msn.com>
```

Stealing patches isn't nice.

**Review by Proximyst — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0662-Workaround-for-visually-missing-health-on-join.patch (hunk 1 above)
Comment (HexedHero) [reply Spigot-Server-Patches/0662-Workaround-for-visually-missing-health-on-join.patch]: Wasn't my intention, I don't know who he was and didn't know his information nor where to even find it and I wouldn't have wanted to publish his email without consent.

Edit: Now I see it's on his profile, didn't realize the author of the issue was the same guy from Spigot

Comment (Proximyst) [reply Spigot-Server-Patches/0662-Workaround-for-visually-missing-health-on-join.patch]: He published it himself. ^^ Thanks for fixing!

**Review by Proximyst — APPROVED** [review]
I'll give this an approval, but I'd like to hear from @MiniDigger and potentially @electronicboy. I'm sure we can get to the bottom of the actual bug with a little bit of work; I know @JRoy was also looking into this earlier.

**Comment by JRoy** [comment]
My main thing was gonna be that this doesn't actually solve the problem and is a nasty workaround. The problem is that **something** in Craftbukkit/Spigot itself is not properly sending the scaled health correctly on the user's initial join (The vanilla server does this correctly). The approach of `yeah let's just try this again and hope it works` isn't a solution that I think is proper here.

**Comment by MiniDigger** [comment]
worth noting that this patch (by a different author) was rejected upstream for the reason jroy mentioned (and I agree that it's not a good solution)

I think the root cause here is that the bukkit representation of the health scale stuff isn't getting initialized. ill propose a different patch once I figured it out

**Comment by Doc94** [comment]
> worth noting that this patch (by a different author) was rejected upstream for the reason jroy mentioned (and I agree that it's not a good solution)
> 
> I think the root cause here is that the bukkit representation of the health scale stuff isn't getting initialized. ill propose a different patch once I figured it out

Yeah, i suggest this patch in bukkit/spigot based in the same "patch" in bukkit for player teleport into diffeten world (when teleport bukkit use the update scale health), actually cant understand the root of this issue for make a property patch and well md_5 cant found the issue to :/

**Comment by JRoy** [comment]
> > worth noting that this patch (by a different author) was rejected upstream for the reason jroy mentioned (and I agree that it's not a good solution)
> > I think the root cause here is that the bukkit representation of the health scale stuff isn't getting initialized. ill propose a different patch once I figured it out
> 
> Yeah, i suggest this patch in bukkit/spigot based in the same "patch" in bukkit for player teleport into diffeten world (when teleport bukkit use the update scale health), actually cant understand the root of this issue for make a property patch and well md_5 cant found the issue to :/

I don't understand what this means, are you using a translator? I have no idea why you're talking about teleports.

**Comment by Proximyst** [comment]
Look at #5139 for a proper fix.

**Pull request closed without merging**